manual_correction.py

```diff
@@ -449,36 +446,37 @@ def check_if_modified(time_one, time_two):
 
 def create_json(fname_nifti, name_rater, modified):
     """
-    Create json sidecar with meta information
-    :param fname_nifti: str: File name of the nifti image to associate with the json sidecar
+    Create/update JSON sidecar with meta information
+    :param fname_nifti: str: File name of the nifti image to associate with the JSON sidecar
     :param name_rater: str: Name of the expert rater
     :param modified: bool: True if the file was modified by the user
     :return:
     """
     if modified:
         metadata = {'Author': name_rater, 'Date': time.strftime('%Y-%m-%d %H:%M:%S')}
         fname_json = fname_nifti.rstrip('.nii').rstrip('.nii.gz') + '.json'
```

Comment:
@NathanMolinier, I just discovered that the `.rstrip` method here is dangerous! For files ending with the letter `n`, for example, `'sub-001_T2w_lesion.nii.gz'`, also the ending letter `n` is stripped:

```python
> 'sub-001_T2w_lesion.nii.gz'.rstrip('.nii').rstrip('.nii.gz') + '.json'
sub-001_T2w_lesio.json
```

We should use `.replace` instead:

```python
> 'sub-001_T2w_lesion.nii.gz'.replace('.nii.gz', '').replace('.nii', '') + '.json'
sub-001_T2w_lesion.json
```

Could you please fix this?

Reply:
Ideally, we should also add a new test(s) into https://github.com/spinalcordtoolbox/manual-correction/blob/main/tests/test_create_json.py

Reply:
Yep I'll do it !

We can actually simply do that

> ```python
> > 'sub-001_T2w_lesion.nii.gz'.replace('.nii.gz', '.json')
> sub-001_T2w_lesion.json
> ```

Reply:
Really? Nifti files might have either `.nii` or `.nii.gz` suffixes. We want to make sure to replace any of them.

Reply:
We can then do that
> 
> ```python
> > 'sub-001_T2w_lesion.nii'.replace('.gz', '').replace('.nii', ''.json'') + 
> sub-001_T2w_lesion.nii.json
> ```

Reply:
> We can then do that
> 
> > ```python
> > > 'sub-001_T2w_lesion.nii'.replace('.gz', '').replace('.nii', '.json')
> > sub-001_T2w_lesion.nii.json
> > ```

Yes, this also works for both `.nii` and `.nii.gz` suffixes.

Reply:
Addressed in https://github.com/spinalcordtoolbox/manual-correction/pull/43/commits/7f942bf3e25a7d57782c27453f60a4548334568b